src/plz/plz_test.go

```diff
@@ -0,0 +1,70 @@
+package plz
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/thought-machine/please/src/cli"
+	"github.com/thought-machine/please/src/core"
+)
+
+func TestStripHost(t *testing.T) {
+	hostArch := cli.HostArch()
+	testData := []struct {
+		Name     string
+		Plugin   string
+		Label    string
+		Expected string
+	}{
+		{
+			Name:     "Leaves no subrepo",
+			Label:    "//foo:bar",
+			Expected: "//foo:bar",
+		},
+		{
+			Name:     "Strips host arch",
+			Label:    fmt.Sprintf("@%v//foo:bar", hostArch.String()),
+			Expected: "//foo:bar",
+		},
+		{
+			Name:     "Leaves target arch",
+			Label:    fmt.Sprintf("@%v//foo:bar", "foo_bar64"),
+			Expected: "@foo_bar64//foo:bar",
+		},
+		{
+			Name:     "Strips plugin repo",
+			Label:    "@go//foo:bar",
+			Plugin:   "go",
+			Expected: "//foo:bar",
+		},
+		{
+			Name:     "Leaves different subrepo",
+			Label:    "@python//foo:bar",
+			Plugin:   "go",
+			Expected: "@python//foo:bar",
+		},
+		{
+			Name:     "Strips host plugin repo",
+			Label:    fmt.Sprintf("@%v//foo:bar", core.SubrepoArchName("go", cli.HostArch())),
+			Plugin:   "go",
+			Expected: "//foo:bar",
+		},
+		{
+			Name:     "Strips host but leaves different subrepo",
+			Label:    fmt.Sprintf("@%v//foo:bar", core.SubrepoArchName("python", cli.HostArch())),
+			Plugin:   "go",
+			Expected: "@python//foo:bar",
+		},
```

Comment:
This might break if we define a subrepo in a plugin that's prefixed with the plugin name e.g. `go_subrepo` in the `go` plugin. I.e. `@go_subrepo//foo:bar` you'd expect to end up with an unchanged label I think? Don't we do something like that in the toolchain?

Reply:
Ah good spot! I don't think that works at the moment from either the command line or when defining dependencies in BUILD files. There hasn't been a need thus far, but maybe this is something we should think about. 

Right now, `@go_subrepo//:all` will try and parse the root package to find a subrepo called `go_subrepo`.